packages/web/modals/lock-tokens.tsx

```diff
@@ -0,0 +1,153 @@
+import Image from "next/image";
+import { FunctionComponent, useState } from "react";
+import { observer } from "mobx-react-lite";
+import { Duration } from "dayjs/plugin/duration";
+import classNames from "classnames";
+import { RatePretty, CoinPretty } from "@keplr-wallet/unit";
+import { ObservableAmountConfig } from "@osmosis-labs/stores";
+import { Button } from "../components/buttons";
+import { InputBox } from "../components/input";
+import { Error } from "../components/alert";
+import { ModalBase, ModalBaseProps } from "./base";
+
+// TODO: add SFS UI
+
+export const LockTokensModal: FunctionComponent<
+  ModalBaseProps & {
+    gauges: { id: string; duration: Duration; apr: RatePretty }[];
+    amountConfig: ObservableAmountConfig;
+    availableToken?: CoinPretty;
+    onLockToken: (gaugeId: string) => void;
+  }
+> = observer((props) => {
+  const { gauges, amountConfig: config, availableToken, onLockToken } = props;
+  const [selectedGaugeIndex, setSelectedGaugeIndex] = useState<number | null>(
+    null
+  );
+
+  return (
+    <ModalBase {...props}>
+      <div className="flex flex-col gap-8 pt-8">
+        <div className="flex flex-col gap-2.5">
+          <span className="subitle1">Unbonding period</span>
+          <div className="flex gap-4">
+            {gauges.map(({ id, duration, apr }, index) => (
+              <LockupItem
+                key={id}
+                duration={duration.humanize()}
+                isSelected={index === selectedGaugeIndex}
+                onSelect={() => setSelectedGaugeIndex(index)}
+                apr={apr.toString()}
+              />
+            ))}
+          </div>
+        </div>
+        <div className="flex flex-col gap-2 border border-white-faint rounded-2xl p-4">
+          <span className="subtitle1">Amount To Bond</span>
+          {availableToken && (
+            <div className="flex gap-1 caption">
+              <span>Available LP Token</span>
+              <span className="text-primary-50">
+                {availableToken.toString()}
+              </span>
+            </div>
+          )}
+          <InputBox
+            currentValue={config.amount}
+            onInput={(value) => config.setAmount(value)}
+            placeholder=""
+            labelButtons={[
+              {
+                label: "MAX",
+                onClick: () => config.setFraction(1),
+                className: "!my-auto !h-6 !bg-primary-200 !caption",
+              },
+            ]}
+          />
+        </div>
+        {config.getError()?.message !== undefined && (
+          <Error
+            className="mx-auto"
+            message={config.getError()?.message ?? ""}
+          />
+        )}
+        <Button
+          className="h-14 w-96 mt-3 mx-auto"
```

Comment:
font size looks smaller than the design in the [figma](https://www.figma.com/file/UXgL2cPiw5I4GALDOodfCy/Osmosis-Web-(Master-File)?node-id=516%3A21569).